rpc/common/moc_common_common.proto

```diff
@@ -60,7 +60,7 @@ enum HealthState {
 
 message Error {
 	string Message = 1;
-	int32 Code = 2;
+	uint32 Code = 2;
```

Comment:
I think this could cause issue during the upgrade. The status field may fail to unmarshal after the upgrade because the old status object is not compatible with the new object.

But given the code field is not used at all right now, I'm not sure how this would impact. I guess we need to add upgrade tests for this scenario.

Reply:
Reason: mainly just because error codes, from my perspective, shouldn't be able to go negative.

Follow-up: yeah, this is persisted. My impression was that marshalling / unmarshalling wouldn't face any changes since we don't use the field (would only face issues if it was used with negative numbers). Since I'll be testing integrating this into 4 different repos independently, I think that would cover marshalling across different versions of status.

I can also include a run of the moc upgrade pipeline, which I think would cover compatibility with the store.

Reply:
From a practical standpoint, since we would never run out of int32 either, I would have left it as is to avoid the question of changing the format. It's not a big deal, so since its already changed its fine to leave as is IMO. 

For this specifically, since there is a question about the resources being loaded, could you please either do a manual test, or cross check that the upgrade pipeline actually validates that resources created before update are correctly loaded after. I would not assume that this specific case would actually cause the MOC upgrade pipeline to break. It might just ignore resources it couldn't load properly or something. I don't know what the behavior would be, so if the upgrade pipeline already covers this and you feel it is sufficient, please let me know. I am not familiar enough with the specifics of that pipeline and process to make a decision one way or another.

Reply:
Yeah my big concern would've been some kind of resource corruption after we upgrade and start cloudagent/nodeagent back up, but there didn't seem to be any issues (most likely because we don't use the field, so it always defaults to 0) and the tests that could pass did.

The caveat here is that the upgrade is currently broken due to an unrelated regression that is causing a 100% failure rate on a particular test, so we're unable to see test progression past that.

Reply:
Double-checked. On cloudagent startup, we load the resource information from the store as plain interface{} (s). The first thing we do with this data is attempt a conversion to (*internal.Resource). If there were issues with this conversion, then cloudagent would just immediately crash since we don't perform it safely (we assume all the resources will be of type resource). So the upgrade pipeline (even though it didn't fully complete), would have caught this I'm fairly certain

Reply:
Sounds good, thanks for checking.

Reply:
definitely would be good to run some upgrade testing. Would this be considered a breaking change between clients using older moc versions and agents using a newer one?

Reply:
It shouldn't be -- we've done extensive testing on different repos and pipelines. All original functions do retain their original functionality. We tested for backwards compatibility thru the moc-upgrade pipeline, along with adding this moc PR to different repos in the AKS HCI flow (e.g., cloud-operator, mgmtappl, cloudagent, etc.) and then making sure the automated pipelines are all green (even with all other repos using the old moc).

Reply:
yep sorry forgot to close this, found this guide which shows this is a safe update:
https://protobuf.dev/programming-guides/proto3/#updating